include/unifex/heap_allocate.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/get_allocator.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/scope_guard.hpp>
+
+#include <memory>
+#include <type_traits>
+
+namespace unifex {
+
+template <typename Operation, typename Allocator>
+class heap_allocated_operation {
+  using allocator_t = typename std::allocator_traits<Allocator>::template rebind_alloc<Operation>;
+
+public:
+  template<typename Sender, typename Receiver>
+  explicit heap_allocated_operation(Sender&& s, Receiver&& r)
+      : allocator_(get_allocator(r)) {
+    using allocator_traits = std::allocator_traits<allocator_t>;
+    Operation* op = allocator_traits::allocate(allocator_, 1);
+    bool constructorSucceeded = false;
+    scope_guard freeOnError = [&]() noexcept {
+      if (!constructorSucceeded) {
+        allocator_traits::deallocate(allocator_, op, 1);
+      }
+    };
+    op_ = ::new (static_cast<void*>(op)) Operation(
+      cpo::connect((Sender&&)s, (Receiver&&)r));
+    constructorSucceeded = true;
+  }
+
+  ~heap_allocated_operation() {
+    op_->~Operation();
+    std::allocator_traits<allocator_t>::deallocate(allocator_, op_, 1);
+  }
+
+  void start() noexcept {
+      cpo::start(*op_);
+  }
```

Comment:
Should this be `friend auto tag_invoke(tag_t<start>, heap_allocated_operation...` ?

Reply:
That would be the other way to implement the start operation.

At the moment, you can still either define a `tag_invoke()` overload or can define a member-function.